Sumo-Logic-Tools/Event_Based_S3_Automation/script.py

```diff
@@ -0,0 +1,265 @@
+2. Enter the Sumo accessId.
+3. Enter the Sumo accessKey.
+https://help.sumologic.com/Manage/Security/Access-Keys can help with the second and third step.
+'''
+class InitialSetupHelper:
+
+	apiEndpointOrderedList = ['https://api.us2.sumologic.com/api', 'https://api.sumologic.com/api', 'https://api.au.sumologic.com/api', 'https://api.eu.sumologic.com/api', 'https://api.de.sumologic.com/api']
+
+	def takeRequiredInputs(self):
+		print("\n WELCOME! THIS SCRIPT WILL AUTOMATE THE EVENT BASED S3 SETUP FOR YOU. IT WILL ADD THE NOTIFICATION BASED MECHANISM ON ALL OF YOUR POLLING SOURCES! \n")
+		print("\nBUT FIRST YOU NEED TO PROVIDE SOME INPUTS \n")
+		print("\nMAKE SURE THAT YOU HAVE RUN AWS CONFIG AND CONFIGURED THE AWS CLI PROPERLY WITH THE CORRECT CREDENTIALS AND REGION \n")
+		print("\nWHICH SUMO API ENDPOINT DO YOU GUYS USE? [1-5]")
+		print("1. ", self.apiEndpointOrderedList[0], "\n2. ", self.apiEndpointOrderedList[1], "\n3. ", self.apiEndpointOrderedList[2], "\n4. ", self.apiEndpointOrderedList[3], "\n5. ", self.apiEndpointOrderedList[4])
+
+		while True:
+			deployment = int(input("Enter your choice: "))
+			if (deployment < 1 or deployment > 5):
+				print("Please select a value between 1 to 5")
+				continue
+			else:
+				break
+
+		base_url = str(self.apiEndpointOrderedList[deployment - 1])
+
+		print("\nEnter you SUMO ACCESS ID AND ACCESS KEY")
+
+		access_id = input("ACCESS ID: ")
+		access_key = input("ACCESS KEY: ")
+
+		return (base_url, access_id, access_key)
+
+
+'''
+Helps in getting the list of sources on which event based S3 has to be set up.
+It does the following tasks:
+1. Get a list of all the collectors.
+2. Get a list of all the sources in every collector.
+3. Extract the list of polling sources on which event-based-S3 can be set up.
+4. Create a map between bucket name and endpoint and return it for further processing. So if there are multiple sources collecting from the same bucket 
+then the map can look like: Map("bucket1" -> ["endpoint1", "endpoint2"], "bucket2" -> ["endpoint3"]).
+'''
+class SumoSetupHelper():
+
+	collector_url = '/v1/collectors/'
+	pollingTypeSources = ['AwsS3Bucket', 'AwsCloudTrailBucket', 'AwsCloudFrontBucket', 'AwsConfigBucket', 'AwsElbBucket', 'AwsS3AuditBucket']
+	mapOfBucketNameToEndpoint = defaultdict(list)
+
+	def __init__(self, base_url, access_id, access_key):
+		self.base_url = base_url
+		self.access_id = access_id
+		self.access_key = access_key
+
+	def calculateMappingOfBucketNameToEndpoint(self):
+		listOfCollectors = self.getListOfCollectors()
+
+		listOfSourcesLink = self.getSourcesLinks(listOfCollectors)
+
+		listOfPollingSources = self.getPollingSources(listOfSourcesLink)
+
+		self.populateMapOfBucketNameToEndpoint(listOfPollingSources)
+
+		return self.mapOfBucketNameToEndpoint
+
+	def getListOfCollectors(self):
+		return self.makeGetRequest(self.collector_url).json()
```

Comment:
this won't work if the customer has > 1000 collectors (the S3 sources might not even be in the first page).

Reply:
Used the same approach as that of https://github.com/SumoLogic/collector-management-client/blob/master/sumo_mgmt.py#L166